packages/react/src/components/Tree/TreeItem.tsx

```diff
@@ -78,20 +90,24 @@ class TreeItem extends AutoControlledComponent<ReactProps<TreeItemProps>, TreeIt
     accessibility: defaultBehavior,
   }
 
+  public state = {
+    open: false,
+  }
+
   handleTitleOverrides = predefinedProps => ({
     onClick: (e, titleProps) => {
       e.preventDefault()
-      this.trySetState({
+      this.setState({
         open: !this.state.open,
       })
+      _.invoke(this.props, 'onOpenChanged', e, this.props)
       _.invoke(predefinedProps, 'onClick', e, titleProps)
     },
   })
 
   renderContent() {
     const { items, title, renderItemTitle } = this.props
```

Comment:
`const { items, title, renderItemTitle, exclusive } = this.props`
and then use the `exclusive` it would be much more readable